lib/spack/llnl/util/filesystem.py

```diff
@@ -196,10 +196,11 @@ def change_sed_delimiter(old_delim, new_delim, *filenames):
 
 def set_install_permissions(path):
     """Set appropriate permissions on the installed file."""
```

Comment:
Can you mention in the docstring that this has no affect on symlinks (or the files that they point to). I think it would also be worth adding the following as a comment (not a docstring):

```
# If this points to a file maintained in a Spack prefix, it is assumed that
# this function will be invoked on the target. If the file is outside a
# Spack-maintained prefix, the permissions should not be modified.
```